enhancements/network/infiniband.md

```diff
@@ -0,0 +1,139 @@
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over InfiniBand
+- Support IP over InfiniBand (IPoIB)
+- Support InfiniBand on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and InfiniBand devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniBand` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
+
+InfiniBand support will only be available on bare metal deployment.
+
+#### Hardware
+
+The following Mellanox NICs will be supported:
+- Mellanox Technologies MT27700 Family [ConnectX-4]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+#### InfiniBand SR-IOV CNI
+
+InfiniBand SR-IOV CNI `ib-sriov-cni` is a 3rd party CNI plugin that moves
+SR-IOV InfiniBand device to pod namespace and configures device attributes
+such as InfiniBand GUID. It will be delivered in a separate image and copied
+to node CNI directory by a DaemonSet.
+
+#### InfiniBand Kubernetes Daemon
+
+InfiniBand Kubernetes Daemon `ib-kubernetes` is a DaemonSet that watches
+Pod object changes and manages network partitioning via InfiniBand subnet
+manager plugins.
+
+#### SR-IOV Operator
+
+SR-IOV Operator API definition will support `infiniBand` or `ib` device type
+in both `SriovNodeNetworkState` and `SriovNodeNetworkPolicy`.
+
+SR-IOV Operator will support generating net-attach-def Custom Resource with
+`ib-sriov-cni` as CNI plugin type.
+
+#### SR-IOV Config Daemon
+
+SR-IOV Config Daemon, a sub component managed by SR-IOV Operator, will be
+enhanced to change device link type to InfiniBand.
+
+#### SR-IOV Network Device Plugin
+
+SR-IOV Network Device Plugin, a sub component managed by SR-IOV Operator,
+supports selecting SR-IOV devices with link type `ether` or `infiniband`.
+
+
+## Design Details
+
+### Test Plan
+
+- e2e tests will be added in SR-IOV Operator for InfiniBand device type
+- e2e tests will be conducted in baremetal environment as part of SR-IOV tests
+- InfiniBand function must work on supported NICs
+
+### Graduation Criteria
+
+##### -> Tech Preview
+
+- SR-IOV InfiniBand device can be provisioned by SR-IOV Operator
+- SR-IOV InfiniBand device can be advertised as extended node resource
+- Pod can be created with SR-IOV InfiniBand device as additional interface
+- SR-IOV document should be updated to include InfiniBand feature
+- Should pass e2e and QE tests
+
+### Upgrade / Downgrade Strategy
+
+This enhancement is an addon feature in SR-IOV Operator, it follows the same
+upgrade/downgrade strategy as was developed in marketplace operator.
+
+## Implementation History
+
+- SR-IOV Operator is in GA support since 4.3
+- RoCE v2 support in SR-IOV Operator is Tech Preview since 4.3
+
+## Infrastructure Needed [optional]
+
+A new github project needs to be created under openshift organization for
+infiniBand SR-IOV CNI.
```

Comment:
So the intention is to fork the project?

Reply:
No, it's the downstream repo for ib-sriov-cni.